docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -1283,72 +1404,71 @@ spring:
 
 This will send a status 431 if size of any request header is greater than 1000 Bytes.
 
-=== The `RewritePath` `GatewayFilter` Factory
+=== The `RequestRateLimiter` `GatewayFilter` Factory
 
-The `RewritePath` `GatewayFilter` factory takes a path `regexp` parameter and a `replacement` parameter.
-This uses Java regular expressions for a flexible way to rewrite the request path.
-The following listing configures a `RewritePath` `GatewayFilter`:
+The `RequestRateLimiter` `GatewayFilter` factory uses a `RateLimiter` implementation to determine if the current request is allowed to proceed. If it is not, a status of `HTTP 429 - Too Many Requests` (by default) is returned.
 
-.application.yml
+This filter takes an optional `keyResolver` parameter and parameters specific to the rate limiter (described later in this section).
+
+`keyResolver` is a bean that implements the `KeyResolver` interface.
+In configuration, reference the bean by name using SpEL.
+`#{@myKeyResolver}` is a SpEL expression that references a bean named `myKeyResolver`.
+The following listing shows the `KeyResolver` interface:
+
+.KeyResolver.java
 ====
-[source,yaml]
+[source,java]
 ----
-spring:
-  cloud:
-    gateway:
-      routes:
-      - id: rewritepath_route
-        uri: https://example.org
-        predicates:
-        - Path=/red/**
-        filters:
-        - RewritePath=/red/?(?<segment>.*), /$\{segment}
+public interface KeyResolver {
+	Mono<String> resolve(ServerWebExchange exchange);
+}
 ----
 ====
 
-For a request path of `/red/blue`, this sets the path to `/blue` before making the downstream request. Note that the `$` should be replaced with `$\` because of the YAML specification.
+The `KeyResolver` interface lets pluggable strategies derive the key for limiting requests.
+In future milestone releases, there will be some `KeyResolver` implementations.
```

Comment:
(Nothing to change, but I wanted to point out that this is one of the few cases where we use the future tense. The rest of the time, we use the indefinite present tense.)